openmdao/solvers/petsc_ksp.py

```diff
@@ -178,15 +184,17 @@ def solve(self, rhs_mat, system, mode):
         options = self.options
         self.mode = mode
 
-        self.ksp.setTolerances(max_it=options['maxiter'],
-                               atol=options['atol'],
-                               rtol=options['rtol'])
-
         unknowns_mat = OrderedDict()
         maxiter = self.options['maxiter']
 
         for voi, rhs in iteritems(rhs_mat):
 
+            ksp = self.ksp[voi]
+
+            ksp.setTolerances(max_it=options['maxiter'],
```

Comment:
a temporary for maxiter is already defined outside of the loop, so no reason to do a lookup for 'maxiter' option again here.